functions/get_status/main.py

```diff
@@ -24,32 +27,36 @@ def __init__(self, bq_errors):
 
 
 def get_status(request):
-    if not request.is_json:
-        return jsonify({"status": "failed", "message": "invalid data"}), 422
-
-    request_data = request.get_json()
-
-    for key in ["user_id", "platform", "os_version", "device_type", "app_version", "lang", "last_beacon_date"]:
-        if key not in request_data:
-            return jsonify({"status": "failed", "message": f"missing field: {key}"}), 422
-
-    user_id = request_data["user_id"]
-    platform = request_data["platform"]
-    os_version = request_data["os_version"]
-    device_type = request_data["device_type"]
-    app_version = request_data["app_version"]
-    lang = request_data["lang"]
-    last_beacon_date = request_data["last_beacon_date"]
-
-    user_entity = _get_user_entity(user_id)
-    if not user_entity:
-        return jsonify({"status": "failed", "message": f"unauthorized"}), 401
+    try:
+        request_data = get_request_data(request)
+        validate_request_parameters(request_data, extra_params=[ExtraParam(KEY_LAST_BEACON_DATE, True)])
+    except InvalidRequestException as e:
+        return jsonify(
+            e.response,
+            e.status
+        )
+
+    user_id = request_data[KEY_USER_ID]
+    platform = request_data[KEY_PLATFORM]
+    os_version = request_data[KEY_OS_VERSION]
+    device_type = request_data[KEY_DEVICE_TYPE]
+    app_version = request_data[KEY_APP_VERSION]
+    lang = request_data[KEY_LANG]
+    last_beacon_date = request_data[KEY_LAST_BEACON_DATE]
+
+    try:
+        user_entity = get_user_from_datastore(user_id)
+    except InvalidRequestException as e:
+        return jsonify(
```

Comment:
Same error as we had in /send_encounters

Should be `return jsonify(e.response), e.status`

Reply:
Thanks ;) rebase-merge issues... fixed